parser.go

```diff
@@ -276,6 +325,27 @@ func (f *File) parse(reader io.Reader) (err error) {
 
 	var line []byte
 	var inUnparseableSection bool
+
+	parserBufferSize := 0
+	// NOTE: Peek 1kb at a time.
+	currentPeekSize := 1024
+
+	// NOTE: Iterate and increase `currentPeekSize` until
+	// the size of the parser buffer is found.
+	// TODO: When Golang 1.10 is the lowest version supported,
+	// replace with `parserBufferSize := p.buf.Size()`.
+	for {
+		peekBytes, _ := p.buf.Peek(currentPeekSize)
+		peekBytesLength := len(peekBytes)
+
+		if parserBufferSize >= peekBytesLength {
+			break
+		}
+
+		currentPeekSize *= 2
+		parserBufferSize = peekBytesLength
+	}
+
```

Comment:
This new block of code can be skipped if `f.options.AllowPythonMultilineValues` is `false`, do you agree?